v2/mongodb-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/mongodb/templates/JavascriptDocumentTransformer.java

```diff
@@ -0,0 +1,267 @@
+/*
```

Comment:
I'm by no means an expert on UDF templates, but it appears as though a lot of this code repeats what is already defined in https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/JavascriptTextTransformer.java

Could this existing class not be used directly? Or at least make JavascriptDocumentTransformer a subclass of JavascriptTextTransformer and override only the necessary methods?

Reply:
I am trying to Transform the Bson Document and the existing code uses the Text transformation. I don't want to break the logic and keep them separate.